File: app/src/main/java/com/example/aleksei/repoinfo/model/api/RetrofitManager.java
package com.example.aleksei.repoinfo.model.api;

import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitManager {

    private static RetrofitManager retrofitManager;
    private static Retrofit retrofit;
    private static final String BASE_URL = "https://api.github.com/";

    private RetrofitManager() {
        retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();
    }

    public static RetrofitManager getInstance() {

        if (retrofitManager == null) {
            retrofitManager = new RetrofitManager();
        }
        return retrofitManager;
    }

    public GitHubApi getJSONApi() {
        return retrofit.create(GitHubApi.class);
    }
}

File: app/src/main/java/com/example/aleksei/repoinfo/model/api/RetrofitManager.kt
package com.example.aleksei.repoinfo.model.api

import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitManager{
    private const val BASE_URL : String = "https://api.github.com/"
    private var retrofit : Retrofit

    init {
        retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build()

    }

    fun getJSONApi() : GitHubApi{
        return retrofit.create(GitHubApi::class.java)
    }
}